pkg/server/sandbox_run_windows.go

```diff
@@ -54,13 +56,30 @@ func (c *criService) RunPodSandbox(ctx context.Context, r *runtime.RunPodSandbox
 	}
 	log.G(ctx).Debugf("Sandbox config %+v", config)
 
-	// Generate unique id and name for the sandbox and reserve the name.
-	id := util.GenerateID()
 	metadata := config.GetMetadata()
 	if metadata == nil {
 		return nil, errors.New("sandbox config must include metadata")
 	}
 	name := makeSandboxName(metadata)
+
+	// check if this is actually a restart
+	id, err := c.sandboxNameIndex.GetKey(name)
+	if err == nil && config.Annotations[annotations.AllowReset] == "true" {
+		log.G(ctx).Infof("reseting existing sandbox %q with id %q", name, id)
+		sandbox, err := c.sandboxStore.Get(id)
+		if err != nil {
+			return nil, errors.Wrapf(err, "an error occurred when try to find sandbox %q", id)
+		}
+		if err = c.resetSandbox(ctx, sandbox); err != nil {
+			return nil, errors.Wrapf(err, "could not reset sandbox %q", id)
```

Comment:
looking back, I dont think its a good idea to overlook the error here, because the reset operation mutates the state for the sandbox (and containers in it), as well as (re)creates resources. 
If we returned a name reservation error, then people might conclude that the operation failed with name reservation nothing happened, when in fact there could be orphaned resources from the reset process if the error came from the cleanup stage.

Reply:
makes sense. I think the real question I should've asked is the expectation around `AllowReset` (now `EnableReset`), whether we should hard fail on it or not. From what you're describing, we should hard fail, since we modify the state.